### PR TITLE
Player Movement

### PR DESCRIPTION
**Pull request by Galaxy25**
#1 
The character can move within the game and its speed can be changed. Interpolation needs to be turned on in order to remove the jittery effect caused by the graphics engine and physics engine from being out of sync. [Interpolation](https://docs.unity3d.com/ScriptReference/Rigidbody-interpolation.html) is recommended by Unity for only the player character because it's more noticeable.

### CODE REVIEW

File: Assets/Scripts/PlayerMovement.cs [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+public class PlayerMovement : MonoBehaviour
+{
+    private Rigidbody2D rb;
+    private float x;
+    private float y;
+    private Vector2 direction;
+    public float speed;
+
+    void Start()
```
Comment (BrandonPacewic) [inline Assets/Scripts/PlayerMovement.cs:13]:
```suggestion
    private void Start()
```

File: Assets/Scripts/PlayerMovement.cs [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    {
+        rb = gameObject.GetComponent<Rigidbody2D>();
+    }
+    // Update is called with frequency of physics engine
+    void FixedUpdate()
```
Comment (BrandonPacewic) [inline Assets/Scripts/PlayerMovement.cs:18]:
```suggestion
    private void FixedUpdate()
```

File: Assets/Scripts/PlayerMovement.cs [hunk 3]
```diff
@@ -0,0 +1,27 @@
+using System.Collections;
+using System.Collections.Generic;
```
Comment (BrandonPacewic) [inline Assets/Scripts/PlayerMovement.cs:2]: Unused import statements.

File: Assets/Scripts/PlayerMovement.cs [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+using UnityEngine;
+
+public class PlayerMovement : MonoBehaviour
+{
+    private Rigidbody2D rb;
+    private float x;
+    private float y;
+    private Vector2 direction;
+    public float speed;
```
Comment (BrandonPacewic) [inline Assets/Scripts/PlayerMovement.cs:11]: Order of variable declarations. Probably best in this case to simply go alphabetically.

**Review by BrandonPacewic — COMMENTED** [review]
General nitpicks. I'll push in a few minutes.

**Comment by BrandonPacewic** [comment]
Closes #1.

File: Assets/Scripts/PlayerMovement.cs (hunk 4 above)
Comment (BrandonPacewic) [reply Assets/Scripts/PlayerMovement.cs]: Also going to start putting all of the public variables on the top. I will make this a consistency thing for all of our files.

**Review by BrandonPacewic — DISMISSED** [review]
Looks good to me! I'll push a Unity version update at some point so whever you get a chance please get the newest version 🙃

**Comment by BrandonPacewic** [comment]
> When using public and private Microsoft want variables to start with a capital and private to start with a lowercase

Really? So then our code would look something like this:

```cs
public double Speed;

private double speed;
```

That does not make any sense to me.

**Comment by Galaxy25** [comment]
https://learn.microsoft.com/en-us/mixed-reality/world-locking-tools/documentation/howtos/codingconventions

https://learn.microsoft.com/en-us/dotnet/csharp/fundamentals/coding-style/coding-conventions

**Review by BrandonPacewic — APPROVED** [review]

**Comment by Galaxy25** [comment]
Speed wasn't changed in the entire script. It should now be working

Approved

**Comment by Galaxy25** [comment]
I'm requesting to merge this branch into main to make it easier to build other features

**Comment by BrandonPacewic** [comment]
> I'm requesting to merge this branch into main to make it easier to build other features

Can do. I wanted to have Jake go through the review process as well but we will need to do this another time.

**Comment by BrandonPacewic** [comment]
Added to the changelog.

**Pull request merged**